### PR TITLE
Add member functions(setNode,getNode) in class Component .

### PR DESCRIPTION
**Pull request by chengstory**
1.  add virtual void setNode(cocos2d::Node *pNode)
2.  add virtual cocos2d::Node\* getNode() const  in class Component .

### CODE REVIEW

File: cocos2dx/support/component/CCComponent.cpp [hunk 1]
```diff
@@ -82,6 +82,15 @@ void Component::setName(const char *pName)
     _name.assign(pName);
 }
 
+void Component::setNode(cocos2d::Node *pNode)
+{
+     CC_UNUSED_PARAM(pNode);
+}
+cocos2d::Node* Component::getNode() const
```
Comment (dumganhar) [inline cocos2dx/support/component/CCComponent.cpp]: It's in cocos2d namespace, so the prefix `cocos2d::` isn't needed.

Comment (chengstory) [inline cocos2dx/support/component/CCComponent.cpp]: ok

File: cocos2dx/support/component/CCComponent.h [hunk 2]
```diff
@@ -48,6 +48,9 @@ class CC_DLL Component : public Object
     const char* getName() const;
     void setName(const char *pName);
     
+    void setNode(cocos2d::Node *pNode);
+    cocos2d::Node* getNode() const;
```
Comment (dumganhar) [inline cocos2dx/support/component/CCComponent.h:52]: `cocos2d::` isn't needed here too.

File: extensions/CocoStudio/Components/CCComRender.cpp [hunk 3]
```diff
@@ -57,7 +57,12 @@ void ComRender::onExit()
     _render = NULL;
 }
 
-cocos2d::Node* ComRender::getNode()
+void ComRender::setNode(cocos2d::Node *pNode)
```
Comment (dumganhar) [inline extensions/CocoStudio/Components/CCComRender.cpp:60]: cocos2d:: isn't needed here too.

File: extensions/CocoStudio/Components/CCComRender.cpp [hunk 4]
```diff
@@ -57,7 +57,12 @@ void ComRender::onExit()
...
+{
+    _render = pNode;
+}
+
+cocos2d::Node* ComRender::getNode() const
```
Comment (dumganhar) [inline extensions/CocoStudio/Components/CCComRender.cpp:65]: cocos2d:: isn't needed here too.

File: extensions/CocoStudio/Components/CCComRender.h [hunk 5]
```diff
@@ -41,7 +41,8 @@ class ComRender : public cocos2d::Component
 public:
    virtual void onEnter();
    virtual void onExit();
-   cocos2d::Node* getNode();
+   void setNode(cocos2d::Node *pNode);
+   cocos2d::Node* getNode() const;
```
Comment (dumganhar) [inline extensions/CocoStudio/Components/CCComRender.h:45]: cocos2d:: isn't needed here too.

Comment (chengstory) [inline extensions/CocoStudio/Components/CCComRender.h:45]: ok

**Pull request closed without merging**